src/main/kotlin/turku/forge/babyproject/api/ChannelController.kt

```diff
@@ -0,0 +1,40 @@
+package turku.forge.babyproject.api
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.messaging.simp.SimpMessagingTemplate
+import org.springframework.web.bind.annotation.*
+import turku.forge.babyproject.config.SIMPLE_MESSAGE_BROKER_ENDPOINT
+import java.util.*
+
+@RestController
+@CrossOrigin(origins = ["*"])
+class ChannelController {
+    @Autowired
+    private lateinit var template: SimpMessagingTemplate
+
+    @PostMapping("$SIMPLE_MESSAGE_BROKER_ENDPOINT/{channelName}", consumes = ["application/json"])
+    fun message(@PathVariable(value = "channelName") channelName: String?, @RequestBody message: Message?): ResponseEntity<*> {
+        message?.let {
+            val destination = "$SIMPLE_MESSAGE_BROKER_ENDPOINT/$channelName"
+            template.convertAndSend(destination, OutputMessage(it))
+            return ResponseEntity(null, HttpStatus.OK)
+        }
+
+        return ResponseEntity(null, HttpStatus.BAD_REQUEST)
+    }
+}
+
+/**
+ * A POJO for messages
```

Comment:
What's a POJO?

Reply:
It stands for `Plain old Java object` I need to still document this stuff a bit better, But this is still a `WIP` I just need some one to take a look at the logic and see if I'm doing anything crazy wrong here